### PR TITLE
add krr page in cookbook

### PR DESCRIPTION
**Pull request by sanuj**
@karlnapf looking forward to your comments :)

### CODE REVIEW

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+=======================
+Kernel Ridge Regression
+=======================
+
+Kernel ridge regression is a kernel-based regularized form of regression which learns a function in the space induced by the respective kernel and the data by minimizing a squared error loss with :math:`L_2` regularization. For linear ridge regression, it boils down to solving a linear system:
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: Kernel ridge regression is a non-parametric form of ridge regression. The aim is to learn a function in the space induced by the respective kernel $k$ by minimizing a squared loss with a squared norm regularization term

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: remove the linear part completely

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    {\bf w} = \left({\bf X}^T{\bf X}+\tau{\bf I}\right)^{-1}{\bf y}
+
+where :math:`{\bf X}` is the feature matrix, :math:`{\bf y}` is the vector of labels and :math:`\tau>0` scales the regularization term.
+
+In case of kernels, a non-linear mapping :math:`\Phi : x \rightarrow \Phi(x) \in \mathcal F` is used. By using kernel trick :math:`K({\bf x_i, x_j})=\Phi({\bf x_i})\Phi({\bf x_j})` the following solution can be obtained:
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: use \mapsto rather than \rightarrow

Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: "from input space into a feature space \cal F is used

Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: please put \langle ... \rangle_\mathcal{F} around the Phi product

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+.. math::
+    {\bf \alpha} = \left({\bf K}+\tau{\bf I}\right)^{-1}{\bf y}
+
+where :math:`{\bf K}` is the kernel matrix and :math:`{\bf \alpha}` is the vector of weights in the space induced by the kernel.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: Say that the learned function can then be evaluated as $f(x)=\sum_{i=1}^Nk(x,x_i)$

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Choose an appropriate :sgclass:`CKernel` and instantiate it. Here we use a :sgclass:`CGaussianKernel`.
+
+.. sgexample:: kernel_ridge_regression.sg:create_appropriate_kernel
+
+We create an instance of :sgclass:`CKernelRidgeRegression` classifier by passing it :math:`\tau`, the kernel and labels.
+
+.. sgexample:: kernel_ridge_regression.sg:create_instance
+
+Then we train the regression model and apply it to test data to get the predicted :sgclass:`CRegressionLabels` and bias.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: KRR doesnt have a bias term

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+.. sgexample:: kernel_ridge_regression.sg:train_and_apply
+
+Imagine, we know the bias term. We can set it as
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: Remove this, KRR should not even offer this method

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst (hunk 3 above)
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: Actually, please remove Phi here completely. We dont need it in an API example, it doesnt matter for that

**Comment by karlnapf** [comment]
I think there also will be a description for krr, which can be removed

**Comment by sanuj** [comment]
@karlnapf I don't fully understand your last comment.
Rest is updated :)

**Comment by karlnapf** [comment]
Check the descriptions folder in the examples dir

**Comment by sanuj** [comment]
@karlnapf removed the description file.

**Comment by karlnapf** [comment]
Cool this is good to merge.
Just a thought I had: Can you also explain how to extract the alpha vector (and the w vector in linear regression)? This will (annoyingly) also change the data files, so we can do it in a separate patch.

But I think it is good to explain such things too.

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+Kernel ridge regression is a non-parametric form of ridge regression. The aim is to learn a function in the space induced by the respective kernel :math:`k` by minimizing a squared loss with a squared norm regularization term.
+
+The following solution can be obtained:
+
+.. math::
+    {\bf \alpha} = \left({\bf K}+\tau{\bf I}\right)^{-1}{\bf y}
+
+where :math:`{\bf K}` is the kernel matrix and :math:`{\bf \alpha}` is the vector of weights in the space induced by the kernel.
+The learned function can then be evaluated as :math:`f(x)=\sum_{i=1}^Nk(x,x_i)`.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: the alpha_i are missing here .. sorry I forgot above

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst [hunk 8]
```diff
@@ -0,0 +1,47 @@
+=======================
+Kernel Ridge Regression
+=======================
+
+Kernel ridge regression is a non-parametric form of ridge regression. The aim is to learn a function in the space induced by the respective kernel :math:`k` by minimizing a squared loss with a squared norm regularization term.
+
+The following solution can be obtained:
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: The solution can be written in closed form as

File: doc/cookbook/source/examples/regression/kernel_ridge_regression.rst (hunk 7 above)
Comment (sanuj) [inline doc/cookbook/source/examples/regression/kernel_ridge_regression.rst]: yeah, i was thinking the same but was not sure. Updating.

**Comment by sanuj** [comment]
@karlnapf Updated

**Pull request merged**